demo/Menu.tsx

```diff
@@ -0,0 +1,70 @@
+import { toggleMark } from "prosemirror-commands";
+import type { MarkType } from "prosemirror-model";
+import type { EditorState } from "prosemirror-state";
+import React, { ReactNode } from "react";
+
+import { useEditorEventCallback, useEditorState } from "../src/index.js";
+
+function isMarkActive(mark: MarkType, state: EditorState): boolean {
+  if (!state.selection) return false;
+  const { from, $from, to, empty } = state.selection;
+  return empty
+    ? !!mark.isInSet(state.storedMarks || $from.marks())
+    : state.doc.rangeHasMark(from, to, mark);
+}
+
+export function Button(props: {
+  children?: ReactNode;
+  isActive: boolean;
+  className?: string;
+  onClick: () => void;
+}) {
+  function handleMouseDown(e: React.MouseEvent) {
+    e.preventDefault();
+    props.onClick();
+  }
+  return (
+    <button
+      className={`button ${props.className} ${props.isActive ? "active" : ""}`}
+      onMouseDown={handleMouseDown}
+    >
+      {props.children}
+    </button>
+  );
+}
+
+export default function Menu() {
+  const state = useEditorState();
+  const { marks } = state.schema;
+
+  const toggleBold = useEditorEventCallback((view) => {
+    if (!view) return;
+    const toggleBoldMark = toggleMark(marks["bold"]);
+    toggleBoldMark(view.state, view.dispatch, view);
+  });
+
+  const toggleItalic = useEditorEventCallback((view) => {
+    if (!view) return;
```

Comment:
i'm trying to better understand the react-prosemirror api's. when would view be `null` here? and is the expectation to simply drop the event?

Reply:
On the first render pass, the view will be null, because it's constructed in a layout effect. That means that we sort of have to type the context as returning an optional view, even though in practice, I think it would always be defined by the time the callback actually runs.

Reply:
Yeah, it's expected that you can just drop the event in this case.

Reply:
thanks for explaining. just throwing it out there: have we considered not firing the callback if `view` is null? this way we could make `view` non-nullable which would make the userland code a little less verbose.

Reply:
I don't think we had.

Reply:
I'd be happy to make a PR if it makes sense!

Reply:
Actually, after poking around I realize that my proposal is somewhat limited. 

> On the first render pass, the view will be null, because it's constructed in a layout effect. 

I'm reading [useEditorView](https://github.com/nytimes/react-prosemirror/blob/main/src/hooks/useEditorView.ts) and [ProseMirrorInner](https://github.com/nytimes/react-prosemirror/blob/main/src/components/ProseMirrorInner.tsx) and learning that the EditorContext value has [nullable props](https://github.com/nytimes/react-prosemirror/blob/main/src/contexts/EditorContext.ts) for both EditorState and EditorView. From what I gather it seems like we are waiting for the editor to mount before [we initialize these values](https://github.com/nytimes/react-prosemirror/blob/main/src/hooks/useEditorView.ts#L105).

Is there something we can do here to guarantee these initial values? Defer [the rendering of the provider and its children](https://github.com/nytimes/react-prosemirror/blob/main/src/components/ProseMirrorInner.tsx#L68)?

It would be wonderful to not have to null-check the return values of all the hooks because of this initial-render case.

Reply:
We can't defer rendering the children because we need the mount point to render so that we can mount the view. It's the tradeoff we made to give the user full control of nesting whatever components they want under `<ProseMirror>` and styling the editing root however they want. We have to actually render the element that sets the mount ref.

Reply:
So, I think we need the context to be nullable values, but we maybe don't need to invoke any callbacks for `useEditorEvent` or `useEditorEffect` unless the view is defined.

Reply:
We can open an issue to discuss! It'd be a non-breaking change to make it always defined, but it'd be a breaking change to have to revert that. So, let's talk about it and be sure!

Reply:
Got it! 

I already have the change on a branch and can open a PR where we can discuss.

I'll follow up elsewhere regarding `useEditorState();` returning a nullable value.

Reply:
🚀